pkg/catalog/catalog.go

```diff
@@ -129,8 +131,67 @@ func getDefaultBuilderImages(client *occlient.Client) ([]CatalogImage, error) {
 		imageStreamTagMap[imageStreamTag.Name] = imageStreamTag
 	}
 
-	var builderImages []CatalogImage
+	builderImages := getBuildersFromImageStreams(imageStreams, imageStreamTagMap)
+
+	return builderImages, nil
+}
 
+// SpliceSupportedTags splits the tags in to fully supported and unsupported tags
+func SpliceSupportedTags(catalogImage CatalogImage) ([]string, []string) {
+
+	var supTag, nonSupTag []string
+	tagMap := createImageMap(catalogImage.imageStreamRef.Spec.Tags)
+	for _, tag := range catalogImage.NonHiddenTags {
+		imageName := tagMap[tag]
+		if isSupportedImage(imageName) {
+			supTag = append(supTag, tag)
+		} else {
+			nonSupTag = append(nonSupTag, tag)
+		}
+	}
+	return supTag, nonSupTag
+}
+
+func createImageMap(tagRefs []imagev1.TagReference) map[string]string {
+	tagMap := make(map[string]string)
+	for _, tagRef := range tagRefs {
+		// TODO: the From here can be of multiple types so we need to handle that
+		imageName := tagRef.From.Name
+		if tagRef.From.Kind == "DockerImage" {
+			urlImageName := strings.SplitN(tagRef.From.Name, "/", 2)[1]
```

Comment:
When running the code, this variable has values like `centos/nginx-110-centos7:latest`, `dotnet/dotnet-20-centos7:latest` which are image names without the URL. The value assigned to the variable `urlImageName` has no URL in it. Would it be better to name it `nonurlImageName` or something like that?

Reply:
There is a lot of potential `tagRef.From.Name` values that are not handled here.
For example are we sure that `tagRef.From.Name` can't be image name in a default Docker format using hub.docker.com like `ubuntu`, or `centos/postgresql-10-centos7`?


This is a temporary code that we will hopefully get rid of soon, but we still need to ba aware of all the limitations.